python/nav/web/maintenance/views.py

```diff
@@ -261,7 +261,11 @@ def edit(request, task_id=None, start_time=None, **_):
     task_form = MaintenanceTaskForm(initial=task_form_initial(task, start_time))
 
     if request.method == 'POST':
-        component_keys = get_component_keys(request.POST)
+        try:
+            component_keys = get_component_keys(request.POST)
+        except ValueError:
+            component_keys = None
+            component_data = dict()
```

Comment:
Just a few lines down, there is an alternate version that gets the data from GET arguments instead, like so:

```python
        component_keys = get_component_keys(request.GET)
```

This is typically reachable from something like `maintenance/new/?netbox=foobar`.
The same issue could be a problem here too, no?  Maybe the `ValueError` should be caught and handled inside `get_component_keys()` instead?